pytorch_lightning/utilities/fetching.py

```diff
@@ -47,22 +47,27 @@ class SimpleDataFetcher(AbstractDataFetcher):
             def fetching_function(self):
                 while True:
                     try:
-                        yield next(self.dataloader_iter), False
+                        return next(self.dataloader_iter), False
```

Comment:
Why this change?
I am having a hard time to undertsand why we have a while loop that will only run for one iteration. It's not clear why this hasn't been replaced by an if statement, or why we didn't just remove the while loop here.

Also it seems that this function now became just semantically equivalent to implement __next__ in a regular iterator, why do we still need this fecthing_function here?